### ISSUE TITLE
Assertion error at model/val_to_term.c:147

### ISSUE THREAD

**Issue by rainoftime** [SRI-CSL/yices2#316]
Hi, for this formula, yices 71391420
~~~~
(set-logic UF)
(declare-sort U 0)
(declare-fun f (U) U)
(declare-fun g (U) U)
(declare-fun h (U) U)
(assert (distinct f g h))
(check-sat)
~~~~
~~~~
yices_smt2 xx.smt2
no mapping or default value assigned to the function
yices_smt2: model/val_to_term.c:147: convert_func: Assertion `0' failed.
~~~~

**Comment by dddejan** [comment]
I think higher order isn't be allowed in SMT-LIB. Somehow it works when changing the logic to QF_UF. Either way, it shouldn't crash.

@rainoftime how did you come up with this formula?

CC @BrunoDutertre

**Comment by rainoftime** [comment]
It is mutated from an instance in CVC4's regression test suits.

**Comment by dddejan** [comment]
> It is mutated from an instance in CVC4's regression test suits.

Yes, this is their test case for higher-order reasoning extensions. This is not standard SMT.

```console
$ cvc4 issue316.smt2 
(error "Parse Error: issue316.smt2:6.24: Cannot apply equalty to functions unless --uf-ho is set.")
```

**Comment by BrunoDutertre** [comment]
Fixed in e12fe84e0ebd40afb714e3b212791c9baaed50a2.
